x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -357,18 +367,101 @@ private void createNewProfile(Subject subject, ActionListener<Profile> listener)
-                    // TODO: replace with actual domain information
                     final VersionedDocument versionedDocument = new VersionedDocument(
                         profileDocument,
                         indexResponse.getPrimaryTerm(),
                         indexResponse.getSeqNo()
                     );
                     listener.onResponse(versionedDocument.toProfile(Set.of()));
-                }, listener::onFailure))
+                }, e -> {
+                    if (e instanceof VersionConflictEngineException) {
+                        // Document already exists with the specified ID, get the document with the ID
+                        // and check whether it is the right profile for the subject
+                        getOrCreateProfileWithBackoff(subject, profileDocument, DEFAULT_BACKOFF.iterator(), listener);
+                    } else {
+                        listener.onFailure(e);
+                    }
+                }))
             )
         );
     }
 
+    // Package private for test
+    void getOrCreateProfileWithBackoff(
+        Subject subject,
+        ProfileDocument profileDocument,
+        Iterator<TimeValue> backoff,
+        ActionListener<Profile> listener
+    ) {
+        getVersionedDocument(profileDocument.uid(), ActionListener.wrap(versionedDocument -> {
+            if (versionedDocument == null) {
+                // Document not found. This can happen if the GET request hits a replica that is still processing the document
+                if (backoff.hasNext()) {
+                    final TimeValue backoffTimeValue = backoff.next();
+                    logger.debug("retrying get profile document [{}] after [{}] backoff", profileDocument.uid(), backoffTimeValue);
+                    client.threadPool()
+                        .schedule(
+                            () -> getOrCreateProfileWithBackoff(subject, profileDocument, backoff, listener),
+                            backoffTimeValue,
+                            ThreadPool.Names.GENERIC
+                        );
+                } else {
+                    // Retry has depleted. This can only happen when the document or the profile index itself gets deleted
+                    // in between requests.
+                    listener.onFailure(
+                        new ElasticsearchException("failed to retrieving profile [{}] after all retries", profileDocument.uid())
+                    );
+                }
+                return;
+            }
+            // Ownership check between the subject and the profile document
+            if (subject.canAccessResourcesOf(versionedDocument.doc.subject())) {
+                // The profile document can be accessed by the subject. It must have just got created by another thread, i.e. racing.
+                // Still need to update it with current auth info before return.
+                updateProfileForActivate(subject, versionedDocument, listener);
```

Comment:
This `if` check basically reuses the existing logic from Authentication class so no additional logic is needed to filter profiles locally.